api/working-with-extensions/continuous-integration.md

```diff
@@ -9,108 +9,58 @@ MetaDescription: Use Continuous Integration for testing Visual Studio Code exten
 
 You can create free projects on [Azure DevOps](https://azure.microsoft.com/services/devops/). This gives you source code hosting, planning boards, building and testing infrastructure, and more. On top of that, you get [10 free parallel jobs](https://azure.microsoft.com/services/devops/pipelines/) for building your projects across all 3 major platforms: Windows, macOS and Linux.
 
-After registering and creating your new project, simply add the following `build.yml` to the root of your extension's repository:
+After registering and creating your new project, simply add the following `azure-pipelines.yml` to the root of your extension's repository. Other than the xvfb setup for Linux, the definition is straight-forward:
 
 ```yaml
-jobs:
-  - job: Windows
-    pool:
-      name: Hosted VS2017
-      demands: npm
-    steps:
-      - task: NodeTool@0
-        displayName: 'Use Node 8.x'
-        inputs:
-          versionSpec: 8.x
-      - task: Npm@1
-        displayName: 'Install dependencies'
-        inputs:
-          verbose: false
-      - task: Npm@1
-        displayName: 'Compile sources'
-        inputs:
-          command: custom
-          verbose: false
-          customCommand: 'run compile'
-      - script: 'node node_modules/vscode/bin/test'
-        displayName: 'Run tests'
-  - job: macOS
-    pool:
-      name: Hosted macOS
-      demands: npm
-    steps:
-      - task: NodeTool@0
-        displayName: 'Use Node 8.x'
-        inputs:
-          versionSpec: 8.x
-      - task: Npm@1
-        displayName: 'Install dependencies'
-        inputs:
-          verbose: false
-      - task: Npm@1
-        displayName: 'Compile sources'
-        inputs:
-          command: custom
-          verbose: false
-          customCommand: 'run compile'
-      - script: 'node node_modules/vscode/bin/test'
-        displayName: 'Run tests'
-  - job: Linux
-    pool:
-      name: Hosted Ubuntu 1604
-      demands: npm
-    steps:
-      - task: NodeTool@0
-        displayName: 'Use Node 8.x'
-        inputs:
-          versionSpec: 8.x
-      - task: Npm@1
-        displayName: 'Install dependencies'
-        inputs:
-          verbose: false
-      - task: Npm@1
-        displayName: 'Compile sources'
-        inputs:
-          command: custom
-          verbose: false
-          customCommand: 'run compile'
-      - script: |
-          set -e
-          /usr/bin/Xvfb :10 -ac >> /tmp/Xvfb.out 2>&1 &
-          disown -ar
-        displayName: 'Start xvfb'
-      - script: 'node node_modules/vscode/bin/test'
-        displayName: 'Run tests'
-        env:
-          DISPLAY: :10
+trigger:
+- master
+
+strategy:
+  matrix:
+    linux:
+      imageName: 'ubuntu-16.04'
+    mac:
+      imageName: 'macos-10.13'
+    windows:
+      imageName: 'vs2017-win2016'
+
+pool:
+  vmImage: $(imageName)
+
+steps:
+
+- task: NodeTool@0
+  inputs:
+    versionSpec: '8.x'
+  displayName: 'Install Node.js'
+
+- bash: |
+    if [ $AGENT_OS == "Linux" ]; then
```

Comment:
We should probably use a [`condition`](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml) instead of scripting it. Something like this:

```
condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'))
```